async/kafka/kafka_test.go

```diff
@@ -315,3 +317,252 @@ func TestConsumer_ConsumeWithoutGroup(t *testing.T) {
 
 	ctx.Done()
 }
+
+type decodingTestData struct {
+	counter                eventCounter
+	msgs                   []*sarama.ConsumerMessage
+	decoder                encoding.DecodeRawFunc
+	dmsgs                  [][]string
+	combinedDecoderVersion int32
+}
+
+type eventCounter struct {
+	messageCount int
+	decodingErr  int
+	resultErr    int
+	claimErr     int
+}
+
+func Test_DecodingMessage(t *testing.T) {
```

Comment:
Is this not a nice candidate for table-driven tests?
We can levewrage the comments as sub test names.